iroh-net/src/net/netmon/actor.rs

```diff
@@ -0,0 +1,257 @@
+            callbacks: Default::default(),
+            callback_token: 0,
+        })
+    }
+
+    pub(super) fn subscribe(&self) -> mpsc::Sender<ActorMessage> {
+        self.actor_sender.clone()
+    }
+
+    pub(super) async fn run(mut self) {
+        const DEBOUNCE: Duration = Duration::from_millis(250);
+
+        let mut last_event = None;
+        let mut debounce_interval = tokio::time::interval(DEBOUNCE);
+        let mut wall_time_interval = tokio::time::interval(POLL_WALL_TIME_INTERVAL);
+
+        loop {
+            tokio::select! {
+                biased;
+                _ = debounce_interval.tick() => {
+                    if let Some(time_jumped) = last_event.take() {
+                        if let Err(err) = self.handle_potential_change(time_jumped).await {
+                            warn!("failed to handle network changes: {:?}", err);
+                        };
+                    }
+                }
+                _ = wall_time_interval.tick() => {
+                    trace!("tick: wall_time_interval");
+                    if self.check_wall_time_advance() {
+                        // Trigger potential change
+                        last_event.replace(true);
+                        debounce_interval.reset_immediately();
+                    }
+                }
+                Ok(_event) = self.mon_receiver.recv_async() => {
+                    trace!("network activity detected");
+                    last_event.replace(false);
+                    debounce_interval.reset_immediately();
+                }
+                Some(msg) = self.actor_receiver.recv() => match msg {
+                    ActorMessage::Subscribe(callback, s) => {
+                        let token = self.next_callback_token();
+                        self.callbacks.insert(token, Arc::new(callback));
+                        s.send(token).ok();
+                    }
+                    ActorMessage::Unsubscribe(token, s) => {
+                        self.callbacks.remove(&token);
+                        s.send(()).ok();
+                    }
+                },
+                else => {
+                    break;
+                }
+            }
+        }
+    }
+
+    fn next_callback_token(&mut self) -> CallbackToken {
+        let token = CallbackToken(self.callback_token);
+        self.callback_token += 1;
+        token
+    }
+
+    async fn handle_potential_change(&mut self, time_jumped: bool) -> Result<()> {
+        info!("potential change");
+
+        let new_state = State::new().await;
+        let old_state = &self.interface_state;
+
+        // No major changes, continue on
+        if !time_jumped && old_state == &new_state {
+            debug!("no changes detected");
+            return Ok(());
+        }
+
+        let mut is_major = is_major_change(old_state, &new_state);
+        // Check for time jumps
+        if !is_major && time_jumped {
+            is_major = true;
+        }
```

Comment:
is this not the same as 
```suggestion
        let is_major = is_major_change(old_state, &new_state) || time_jumped;
```